tools/scylla-sstable.cc

```diff
@@ -2234,8 +2234,35 @@ void upgrade_operation(schema_ptr schema, reader_permit permit, const std::vecto
     }
 }
 
+void dump_if_user_type(const data_type& t) {
+    if (t->is_user_type()) {
+        const auto udt = dynamic_pointer_cast<const user_type_impl>(t);
+
+        for (const auto& field_udt : udt->get_all_referenced_user_types()) {
+            dump_if_user_type(field_udt);
```

Comment:
This code can find the same UDT name multiple times and print it multiple times. If I understand the test correctly, the types are printed as CREATE TYPE IF NOT EXISTS so it's safe to print the same one multiple times. Am I right?